main.go

```diff
@@ -382,6 +375,35 @@ func singleflightDo[V any](group *singleflight.Group, key string, fn func() (V,
 	return out.(V), err, shared
 }
 
+// passthroughHandler is an HTTP handler that passes through GET requests to the CT log.
+type passthroughHandler struct {
+	logURL string
+}
+
+func (p passthroughHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "GET" {
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		fmt.Fprintln(w, "only GET is supported")
+		return
+	}
+	url := fmt.Sprintf("%s%s", p.logURL, r.URL.Path)
+	r, err := http.NewRequestWithContext(r.Context(), http.MethodGet, url, nil)
```

Comment:
You are reusing the input `r *http.Request` here.

```suggestion
	req, err := http.NewRequestWithContext(r.Context(), http.MethodGet, url, nil)
```